app/controllers/parties_controller.rb

```diff
@@ -0,0 +1,44 @@
+class PartiesController < ApplicationController
+  def new
+    @id = params[:external_movie_id]
+    @title = params[:title]
+    @duration = params[:duration]
+    @party = Party.new
+  end
+
+  def create
+    new_party = current_user.parties.new(party_params)
+    if params[:party][:duration] < params[:runtime]
+      flash[:error] = 'Error: Party duration must match or exceed movie runtime.'
+      redirect_with_params
+    elsif params[:party][:invitations].nil? || params[:party][:invitations].all? {|friend| friend == ""}
+      flash[:error] = 'Error: Party must need friends.'
+      redirect_with_params
+    elsif new_party.save
+      create_invitations
+     flash[:success] = 'New Viewing Party Created'
+     redirect_to dashboard_path
+    else
+      flash[:error] = 'Error: Party not created'
+      redirect_with_params
+    end
+  end
```

Comment:
This looks way better, but definitely still some opportunity for future refactoring to clean up the controller a bit.